src/gtk/desktopenv.cpp

```diff
@@ -0,0 +1,50 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/desktopenv.cpp
+// Purpose:     implementation of wxDesktopEnvBase
+// Author:      Igor Korot
+// Created:     2021-09-05
+// Copyright:   (c) 2021 wxWidgets development team
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#include <gio/gio.h>
+#include "wx/translation.h"
+#include "wx/log.h"
+#include "wx/desktopenv.h"
+
+// ============================================================================
+// implementation
+// ============================================================================
+
+/* static */
+bool wxDesktopEnv::MoveToRecycleBin(const wxString &path)
+{
+    bool result = false;
+    if( wxDesktopEnvBase::MoveToRecycleBin( path ) )
+    {
+        GError *error = NULL;
+        GFile *f = g_file_new_for_path( path.c_str() );
+        gboolean res = g_file_trash( f, NULL, &error );
+        if( error && error->code == G_IO_ERROR_NOT_SUPPORTED )
+            wxLogSysError( _( "MoveToRecycleBin: Operation not supported." ) );
+        else if( !res )
+            wxLogSysError( _( "Failed to move '%s' to Recycle Bin" ), path );
+        else
+        {
+            result = true;
+            g_object_unref( f );
```

Comment:
Does `g_file_trash()` take ownership of it on success? I.e. are you sure it doesn't need to be unref'd if it fails? I can't find documentation for this function but it would be surprising if it did this, so I think you need to use `wxGtkObject` to avoid leaks here.